src/hooks/useTodo.ts

```diff
@@ -0,0 +1,50 @@
+import { useState, useEffect } from 'react';
+import { Todo } from '../types/Todo';
+import { FilterStatusType } from '../types/FilterStatusType';
+import { getTodos } from '../api/todos';
+import { ErrorMessage } from '../types/ErrorStatusType';
+import { getVisibleTodos } from '../utils/getVisibleTodos';
+
+export function useTodo() {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessage | null>(null);
+  const [filterStatus, setFilterStatus] = useState(FilterStatusType.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [todoIdsToDelete, setTodoIdsToDelete] = useState<number[]>([]);
+  const [todoIdsToUpdate, setTodoIdsToUpdate] = useState<number[]>([]);
```

Comment:
what difference between this states? i believe we should use just one state like processingTodoIds or smth like this
Let's simplify
KISS DRY ;)